### PR TITLE
Feature - migrate tests to Jest

### LINKED ISSUE #149: Migrate tests to Jest
We're currently using Ava, but it lacks many functionalities that Jest does support, including mocks.

Instead of finding a mocks library, I prefer migrating to Jest which has those functionalities in addition to being well-known and much more popular than Ava.

### PR DESCRIPTION
**Pull request by SnirShechter**
Closes #149

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const config: Config.InitialOptions = {
+	preset: 'ts-jest',
+	testEnvironment: 'node',
+	moduleNameMapper: {
+		'^(.*)\\.js$': '$1',
+	},
+	coverageDirectory: 'coverage',
+	coverageProvider: 'v8',
+	collectCoverageFrom: ['src/**/*'],
+	clearMocks: false,
```
Comment (shay123g) [inline jest.config.ts:12]: dont we want to start each test fresh?

Comment (shay123g) [reply jest.config.ts:12]: mean clearMocks:true?

Comment (SnirShechter) [reply jest.config.ts:12]: Yes. Why not?

Comment (shay123g) [reply jest.config.ts:12]: It says clearMocks:false

Comment (SnirShechter) [reply jest.config.ts:12]: You're absolutely correct. I'm not sure why I did that. Fixed.

**Review by shay123g — APPROVED** [review]

**Pull request merged**